forge-game/src/main/java/forge/game/cost/CostAdjustment.java

```diff
@@ -526,64 +526,71 @@ private static boolean checkRequirement(final SpellAbility sa, final StaticAbili
-                if (!sa.isActivatedAbility() || sa.isReplacementAbility()) {
-                    return false;
-                }
-                if (st.hasParam("OnlyFirstActivation")) {
-                    int times = 0;
-                    for (IndividualCostPaymentInstance i : game.costPaymentStack) {
-                        SpellAbility paymentSa = i.getPayment().getAbility();
-                        if (paymentSa.isActivatedAbility() && st.matchesValidParam("ValidCard", paymentSa.getHostCard())) {
-                            times++;
-                            if (times > 1) {
-                                return false;
+                case "Ability" -> {
+                    if (!sa.isActivatedAbility() || sa.isReplacementAbility()) {
+                        return false;
+                    }
+                    if (st.hasParam("OnlyFirstActivation")) {
+                        int times = 0;
+                        for (IndividualCostPaymentInstance i : game.costPaymentStack) {
+                            SpellAbility paymentSa = i.getPayment().getAbility();
+                            if (paymentSa.isActivatedAbility() && st.matchesValidParam("ValidCard", paymentSa.getHostCard())) {
+                                times++;
+                                if (times > 1) {
+                                    return false;
+                                }
                             }
                         }
                     }
                 }
-            } else if (type.equals("NonManaAbility")) {
-                if (!sa.isActivatedAbility() || sa.isManaAbility() || sa.isReplacementAbility()) {
-                    return false;
+                case "NonManaAbility" -> {
+                    if (!sa.isActivatedAbility() || sa.isManaAbility() || sa.isReplacementAbility()) {
+                        return false;
+                    }
                 }
-            } else if (type.equals("MorphDown")) {
-                if (!sa.isSpell() || !sa.isCastFaceDown()) {
-                    return false;
+                case "MorphDown" -> {
+                    if (!sa.isSpell() || !sa.isCastFaceDown()) {
+                        return false;
+                    }
                 }
-            } else if (type.equals("Foretell")) {
-                if (!sa.isForetelling()) {
-                    return false;
+                case "Foretell" -> {
+                    if (!sa.isForetelling()) {
+                        return false;
+                    }
+                    if (st.hasParam("FirstForetell") && activator.getNumForetoldThisTurn() > 0) {
+                        return false;
+                    }
                 }
-                if (st.hasParam("FirstForetell") && activator.getNumForetoldThisTurn() > 0) {
-                    return false;
+                case "TurnFaceUp" -> {
+                    if (!sa.isTurnFaceUp()) return false;
```

Comment:
why new type?
I'd rather see you use `ValidSpell$ Static.isTurnFaceUp`

Reply:
I didn't realize that would work 😆

Reply:
syntax still wrong...